utility/block.go

```diff
@@ -17,26 +17,32 @@ DeliverTx individually applys each transaction against the state and rolls it ba
 like BeginBlock, EndBlock is an autonomous state oepration that executes at the end of every block.
 */
 
-func (u *UtilityContext) ApplyBlock(latestHeight int64, proposerAddress []byte, transactions [][]byte, lastBlockByzantineValidators [][]byte) ([]byte, error) {
+func (u *UtilityContext) ApplyProposalTransactions(latestHeight int64, proposerAddress []byte, transactions [][]byte, lastBlockByzantineValidators [][]byte) ([]byte, error) {
```

Comment:
I'm not sold on this name change. The lifecycle includes 'begin_block' and 'end_block' which is completely separate from 'transactions'

Reply:
Reverting back to ApplyBlock.